backend/open_api_spec.yml

```diff
@@ -0,0 +1,829 @@
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    patch:
+      description: Update a Dataset's curator-provided tag.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#components/parameters/optional_query_dataset_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                  description: curator-provided tag
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-links:
+    put:
+      description: Upload a Dataset. The Dataset will be downloaded from the provided link.
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                id:
+                  $ref: "#/components/schemas/dataset_uuid"
+                link:
+                  $ref: "#/components/schemas/upload_link"
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-s3:
+    put:
+      description: >
+        Retrieve AWS credentials for uploading Dataset source files to S3 to create
+        or update Datasets for the specified Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties: false
+                properties:
+                  credentials:
+                    type: object
+                    properties:
+                      access_key:
+                        type: string
+                        required: true
+                      secret_access_key:
+                        type: string
+                        required: true
+                      access_token:
+                        type: string
+                        required: true
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
```

Comment:
when would we return a 404?

Reply:
will delete to reduce complexity

Reply:
So you agreed that we could leave the 404 for `/curation/collections/{collection_id}/datasets`, which sort of implies we might want to leave it for this route as well _for super curators only_.

What do you think?

Reply:
Hm. I definitely don't agree that we couldn't possibly need a 404 for _any_ reason (see my messages immediately above; if the collection doesn't exist and the curator is a super-user). 

But I would agree it isn't an important part of the response outline.

Reply:
removed (`51f8fa7f`)